public/app/features/datasources/DataSourcesList.tsx

```diff
@@ -1,39 +1,50 @@
 // Libraries
-import React, { PureComponent } from 'react';
-import classNames from 'classnames';
-
-// Components
-import DataSourcesListItem from './DataSourcesListItem';
+import React, { FC } from 'react';
 
 // Types
 import { DataSourceSettings } from '@grafana/data';
-import { LayoutMode, LayoutModes } from '../../core/components/LayoutSelector/LayoutSelector';
+import { LayoutMode } from '../../core/components/LayoutSelector/LayoutSelector';
+import { Card, Tag, useStyles } from '@grafana/ui';
+import { css } from 'emotion';
 
 export interface Props {
   dataSources: DataSourceSettings[];
   layoutMode: LayoutMode;
 }
 
-export class DataSourcesList extends PureComponent<Props> {
-  render() {
-    const { dataSources, layoutMode } = this.props;
-
-    const listStyle = classNames({
-      'card-section': true,
-      'card-list-layout-grid': layoutMode === LayoutModes.Grid,
-      'card-list-layout-list': layoutMode === LayoutModes.List,
-    });
+export const DataSourcesList: FC<Props> = ({ dataSources, layoutMode }) => {
+  const styles = useStyles(getStyles);
 
-    return (
-      <section className={listStyle}>
-        <ol className="card-list">
-          {dataSources.map((dataSource, index) => {
-            return <DataSourcesListItem dataSource={dataSource} key={`${dataSource.id}-${index}`} />;
-          })}
-        </ol>
-      </section>
-    );
-  }
-}
+  return (
+    <ul className={styles.list}>
+      {dataSources.map((dataSource, index) => {
+        return (
+          <li key={`${dataSource.id}-${index}`}>
```

Comment:
fwiw, there's no value in concatenating these together - ``${dataSource.id}-${index}`` is essentially the same as `index` as the non-stable index will "poison" the whole key.

if `dataSource.id` is guarenteed to be unique and stable then just that'll do :)

Reply:
Good catch!